### ISSUE TITLE
The location is not updated when using set_location()

### ISSUE THREAD

**Issue by emmaw19Xmas** [appium/appium#16370]
## The problem

When using the set_location() method, the logs send me 200, however, the mock location is not really be updated on my device.

## Environment

* Appium version: 1.22.2
* Desktop OS/version used to run Appium: Windows 10
* Node.js version: 16.13.2
* Npm or Yarn package manager: 8.1.2
* Mobile platform/version under test: Android samsung Tab A (Android 10)
* Real device or emulator/simulator: Real device

## Details

My testing app has Google Map nested. So I tried to set_location by using Appium. However, when I run the command either from Pycharm or from Appium Inspector, no location updates are shown on my map. 

`driver.set_location(lat, lng)`

## Appium logs

> [MJSONWP (44b50f3d)] Calling AppiumDriver.setGeoLocation() with args: [{"latitude":42.95,"longitude":-79.85,"altitude":0},"44b50f3d-f490-4bd9-a4e7-02857c272218"]
> [ADB] Running 'C:\Users\Emma Wang\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s R9WN20KR6AJ shell am startservice -e longitude -79.85 -e latitude 42.95 -e altitude 0 io.appium.settings/.LocationService'
> [ADB] Running 'C:\Users\Emma Wang\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s R9WN20KR6AJ shell am broadcast -n io.appium.settings/.receivers.LocationInfoReceiver -a io.appium.settings.location'
> [ADB] Got geo coordinates: {"latitude":"42.9345339","longitude":"-78.7437248","altitude":"176.3999939"}
> [MJSONWP (44b50f3d)] Responding to client with driver.setGeoLocation() result: {"latitude":42.9345339,"longitude":-78.7437248,"altitude":176.3999939}
> [HTTP] <-- POST /wd/hub/session/44b50f3d-f490-4bd9-a4e7-02857c272218/location 200 242 ms - 142
> 

## Code To Reproduce Issue [ Good To Have ]

```
def run_trips(file, driver, interval):
    points = pd.read_csv(file, header=0)
    for i in range(0, len(points)):
        lng = points.iloc[i, 0]
        lat = points.iloc[i, 1]
        print(f"{lng}, {lat}, test")
        driver.set_location(lat, lng)
        sleep(interval)

```

**Comment by mykola-mokhnach** [comment]
Duplicate of https://github.com/appium/appium/issues/16341

**Comment by emmaw19Xmas** [comment]
Thanks!  I checked the workaround you posted under #16341, it is helpful and reminds me to check the location GPS mode of settings. 

So here is another method using the ADB command, it works for me.  

Before I set the location with Appium, my device was under the **high accuracy mode**. I checked it with the command below. 
`adb shell settings get secure location_providers_allowed`

After I set the location with Appium, the mock location was not updated on my devices. When I used the command above, no result showed up, It means the location GPS was turned off. Then I used the commands below to turn the service on and also with high accuracy.
```
adb shell settings put secure location_providers_allowed +network 
adb shell settings put secure location_providers_allowed +gps
```
Now my scripts can run very well. And the mock locations can be updated very fast and accurately.
